src/slider/slider.js

```diff
@@ -5,200 +5,135 @@ This source code is licensed under the MIT license found in the
 LICENSE file in the root directory of this source tree.
 */
 // @flow
+
 import * as React from 'react';
-import {findDOMNode} from 'react-dom';
-import {thumbWidth} from './constants.js';
-import type {StatelessStateT, PropsT} from './types.js';
+import {Range} from 'react-range';
+import type {PropsT} from './types.js';
 import {
   Root as StyledRoot,
-  Axis as StyledAxis,
-  AxisRange as StyledAxisRange,
+  Track as StyledTrack,
+  InnerTrack as StyledInnerTrack,
   Tick as StyledTick,
   TickBar as StyledTickBar,
   Thumb as StyledThumb,
+  InnerThumb as StyledInnerThumb,
+  ThumbValue as StyledThumbValue,
 } from './styled-components.js';
 import {getOverrides} from '../helpers/overrides.js';
-/* global document */
-/* global window */
-
-function getTickValue(tick: number | {value: number}): number {
-  if (typeof tick === 'object') {
-    return tick.value;
-  }
-
-  return tick;
-}
-
-function getTickLabel(tick: number | {label: React.Node}): React.Node {
-  if (typeof tick === 'object') {
-    return tick.label;
-  }
-  return tick;
-}
 
-class Slider extends React.Component<PropsT, StatelessStateT> {
-  domNode: ?Element | Text;
+// values.length should not be bigger than two
+// because our design doesn't support more than
+// two thumbs
+const limitValues = (values: number[]): number[] =>
```

Comment:
should throw an exception if more than 2 values are supported, just to make sure folks instantly know why it does not work